### PR TITLE
Remove .NET MAUI internals usage

### PR DESCRIPTION
**Pull request by jfversluis**
## Summary
- Replace MAUI internal FontElement/TextElement/ImageElement/PlaceholderElement/PaddingElement usage with public concrete BindableProperty mappings.
- Narrow dynamic-resource helpers to public Element APIs instead of internal IDynamicResourceHandler/BindableObject internals.
- Replace the custom internal TypedBindingBase-derived implementation with public Binding-based typed binding behavior.
- Add typed-binding regression coverage for read-only/default TwoWay targets, setter write-back, OneWayToSource initial and late BindingContext pushes, object-valued paths, Binding.DoNothing, invalid convert-back values, and captured-value binding contexts.

## Validation
- Product-code internals scan: no matches for Microsoft.Maui.Controls.Internals, .Internals, FontElement, TextElement, ImageElement, PlaceholderElement, PaddingElement, IDynamicResourceHandler, TypedBindingBase, BindingExpression, SetValueCore, GetContext, ApplyCore, UnapplyCore, GetRealizedMode, BindingExpressionPart, BindingFlags.NonPublic, GetMethod/GetField/GetProperty in src/CommunityToolkit.Maui.Markup.
- Unit tests: 402/402 passed via the NUnit executable runner.
- Sample app: net10.0-windows10.0.19041.0 Debug build succeeded.
- Pack: CommunityToolkit.Maui.Markup.99.0.4-internalsfix.nupkg built locally.
- MAUI 11 PR #34070 smoke app: built with Microsoft.Maui.Controls 11.0.0-ci.pr34070.6.26324.9 and launched on Windows; app stayed running after startup with no FieldAccessException/startup crash.

## Notes
- `TypedBinding.cs` was removed because public `BindingBase` cannot be subclassed equivalently outside MAUI internals. The replacement preserves common path binding behavior and explicit setter write-back using public APIs.
- Dynamic-resource helpers now require `Element`, matching the public MAUI dynamic-resource API surface.

## Updated Documentation

The `CommunityToolkit.Maui.Markup` docs have been updated accordingly: https://github.com/MicrosoftDocs/CommunityToolkit/pull/652

### CODE REVIEW

File: src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs [hunk 1]
```diff
@@ -186,17 +191,279 @@ public static TBindable Bind<TBindable, TBindingContext, TSource, TParam, TDest>
...
+			bindable.SetValue(sourceUpdateHandlersProperty, handlers);
+		}
+
+		handlers[targetProperty] = new SourceUpdateHandlers(propertyChangedHandler, bindingContextChangedHandler);
+		bindable.PropertyChanged += propertyChangedHandler;
+		if (bindingContextChangedHandler is not null)
+		{
+			bindable.BindingContextChanged += bindingContextChangedHandler;
+		}
+	}
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs:493]: Addressed by adding `RemoveTypedBinding(...)`, which removes typed-binding source write-back handlers before calling `RemoveBinding(...)`. The XML remarks explicitly direct consumers to use `RemoveTypedBinding` instead of `BindableObject.RemoveBinding(...)` for typed bindings created with setters.

File: src/CommunityToolkit.Maui.Markup/ElementExtensions.cs [hunk 2]
```diff
@@ -1,65 +1,121 @@
...
+	public static TLabel Padding<TLabel>(this TLabel paddingElement, Thickness padding, Label? overload = null) where TLabel : Label => SetPadding(paddingElement, padding);
+
+	/// <inheritdoc />
+	public static TLayout Padding<TLayout>(this TLayout paddingElement, Thickness padding, Layout? overload = null) where TLayout : Layout => SetPadding(paddingElement, padding);
+
+	/// <inheritdoc />
+	public static TPage Padding<TPage>(this TPage paddingElement, Thickness padding, Page? overload = null) where TPage : Page => SetPadding(paddingElement, padding);
+
+	/// <inheritdoc />
+	public static TScrollView Padding<TScrollView>(this TScrollView paddingElement, Thickness padding, ScrollView? overload = null) where TScrollView : ScrollView => SetPadding(paddingElement, padding);
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/ElementExtensions.cs]: I kept the `ScrollView` overload intentionally. The current overload set is not ambiguous for direct `ScrollView.Padding(new Thickness(...))` calls, and the unit tests cover direct `ScrollView` and derived `ScrollView` receivers. Removing the `ScrollView` overload was tested locally and caused the direct `ScrollView` calls to stop compiling instead of falling back to the `Layout` overload.

File: src/CommunityToolkit.Maui.Markup/ElementExtensions.cs [hunk 3]
```diff
@@ -1,65 +1,121 @@
...
+	public static TLabel Padding<TLabel>(this TLabel paddingElement, double horizontalSize, double verticalSize, Label? overload = null) where TLabel : Label => SetPadding(paddingElement, new Thickness(horizontalSize, verticalSize));
+
+	/// <inheritdoc />
+	public static TLayout Padding<TLayout>(this TLayout paddingElement, double horizontalSize, double verticalSize, Layout? overload = null) where TLayout : Layout => SetPadding(paddingElement, new Thickness(horizontalSize, verticalSize));
+
+	/// <inheritdoc />
+	public static TPage Padding<TPage>(this TPage paddingElement, double horizontalSize, double verticalSize, Page? overload = null) where TPage : Page => SetPadding(paddingElement, new Thickness(horizontalSize, verticalSize));
+
+	/// <inheritdoc />
+	public static TScrollView Padding<TScrollView>(this TScrollView paddingElement, double horizontalSize, double verticalSize, ScrollView? overload = null) where TScrollView : ScrollView => SetPadding(paddingElement, new Thickness(horizontalSize, verticalSize));
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/ElementExtensions.cs]: Same as above: this direct `ScrollView.Padding(horizontal, vertical)` call shape is covered by the padding tests and compiles with the current overload set. Removing the `ScrollView` overload regressed compile-time support for direct `ScrollView` calls.

File: src/CommunityToolkit.Maui.Markup/ElementExtensions.cs [hunk 4]
```diff
@@ -1,65 +1,121 @@
...
+	public static TLabel Paddings<TLabel>(this TLabel paddingElement, double left = 0, double top = 0, double right = 0, double bottom = 0, Label? overload = null) where TLabel : Label => SetPadding(paddingElement, new Thickness(left, top, right, bottom));
+
+	/// <inheritdoc />
+	public static TLayout Paddings<TLayout>(this TLayout paddingElement, double left = 0, double top = 0, double right = 0, double bottom = 0, Layout? overload = null) where TLayout : Layout => SetPadding(paddingElement, new Thickness(left, top, right, bottom));
+
+	/// <inheritdoc />
+	public static TPage Paddings<TPage>(this TPage paddingElement, double left = 0, double top = 0, double right = 0, double bottom = 0, Page? overload = null) where TPage : Page => SetPadding(paddingElement, new Thickness(left, top, right, bottom));
+
+	/// <inheritdoc />
+	public static TScrollView Paddings<TScrollView>(this TScrollView paddingElement, double left = 0, double top = 0, double right = 0, double bottom = 0, ScrollView? overload = null) where TScrollView : ScrollView => SetPadding(paddingElement, new Thickness(left, top, right, bottom));
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/ElementExtensions.cs]: Same as above: `ScrollView.Paddings(...)` and derived `ScrollView` receivers are covered by unit tests. The dedicated overload preserves compile-time support without broadening the API to unsupported `BindableObject` types.

File: src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs [hunk 5]
```diff
@@ -186,17 +191,292 @@ public static TBindable Bind<TBindable, TBindingContext, TSource, TParam, TDest>
...
+
+	static bool TryConvertTargetValue<TSource, TParam>(object? targetValue, IValueConverter? converter, TParam? converterParameter, out TSource? sourceValue)
+	{
+		try
+		{
+			var convertedValue = converter?.ConvertBack(targetValue, typeof(TSource), converterParameter, System.Globalization.CultureInfo.CurrentUICulture) ?? targetValue;
+			if (ReferenceEquals(convertedValue, BindableProperty.UnsetValue) || ReferenceEquals(convertedValue, Binding.DoNothing))
+			{
+				sourceValue = default;
+				return false;
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs]: Addressed in `2cc6860` by tracking target values produced by source-to-target binding conversion and consuming the matching `PropertyChanged` notification without invoking `ConvertBack`/setter write-back. Added `SourceUpdateWithNonReversibleConverterDoesNotWriteBackToSource` to cover the non-reversible converter case.

File: src/CommunityToolkit.Maui.Markup/ElementExtensions.cs [hunk 6]
```diff
@@ -1,65 +1,121 @@
...
-	/// <returns>Layout with added Padding</returns>
-	public static TLayout Padding<TLayout>(this TLayout paddingElement, Thickness padding) where TLayout : Element, IPaddingElement
-	{
-		paddingElement.SetValue(PaddingElement.PaddingProperty, padding);
+	/// <summary>Set Padding.</summary>
+	/// <param name="paddingElement">Element on which to set padding.</param>
+	/// <param name="padding">Padding to set.</param>
+	/// <param name="overload">Unused overload discriminator.</param>
+	/// <returns>Element with added Padding.</returns>
+	public static TBorder Padding<TBorder>(this TBorder paddingElement, Thickness padding, Border? overload = null) where TBorder : Border => SetPadding(paddingElement, padding);
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/ElementExtensions.cs]: I am not taking the broad `BindableObject` overload approach because it reintroduces the earlier reviewed problem: unsupported receivers would compile and then fail at runtime. The current overload set is intentionally limited to supported public MAUI types so unsupported padding calls remain compile-time errors while avoiding MAUI internals/private APIs.

**Comment by jfversluis** [comment]
@TheCodeTraveler this PR is now rebased on latest `main` and remains green locally. Could you please prioritize the final review/merge sooner rather than later? We are getting close to merging the .NET MAUI 11 change that removes the internals access this PR fixes for Maui.Markup.

File: .github/workflows/dotnet-build.yml [hunk 7]
```diff
@@ -26,7 +26,7 @@ env:
   PathToCommunityToolkitSampleCsproj: 'samples/CommunityToolkit.Maui.Markup.Sample/CommunityToolkit.Maui.Markup.Sample.csproj'
   PathToCommunityToolkitUnitTestCsproj: 'src/CommunityToolkit.Maui.Markup.UnitTests/CommunityToolkit.Maui.Markup.UnitTests.csproj'
   PathToCommunityToolkitSourceGeneratorsCsproj: 'src/CommunityToolkit.Maui.Markup.SourceGenerators/CommunityToolkit.Maui.Markup.SourceGenerators.csproj'
-  Xcode_Version: '^26'
+  Xcode_Version: '26.6'
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply .github/workflows/dotnet-build.yml]: Changed the workflow back to a floating `^26` selector in `e4e2494`, so CI still selects a 26.x Xcode version required by the current .NET iOS/MacCatalyst workload without depending on one exact patch build being present on every `macos-26` image.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs [hunk 8]
```diff
@@ -186,17 +191,354 @@ public static TBindable Bind<TBindable, TBindingContext, TSource, TParam, TDest>
...
+				var bindingConverter = path is null
+					? new GetterValueConverter<TBindingContext, TSource>(getter, converter, targetUpdateTracker)
+					: targetUpdateTracker is null ? converter : new TargetUpdateTrackingConverter(converter, targetUpdateTracker);
+				var binding = new Binding(
+					path ?? Binding.SelfPath,
+					GetTargetUpdateBindingMode(resolvedMode),
+					bindingConverter,
+					converterParameter,
+					stringFormat,
+					source)
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs]: Fixed in `03964ad`: the target-update tracking converters now apply `stringFormat`, `TargetNullValue`, and `FallbackValue` themselves before tracking, and the binding only receives those options directly when tracking is not needed. I also added regression tests for all three cases to ensure source-driven target updates do not invoke `ConvertBack` or write back to the source.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/TypedGesturesExtensions.cs [hunk 9]
```diff
@@ -261,12 +261,43 @@ public static TGestureElement BindTapGesture<TGestureElement, TCommandBindingCon
...
 	static Func<TBindingContext, TSource> ConvertExpressionToFunc<TBindingContext, TSource>(in Expression<Func<TBindingContext, TSource>> expression) => expression.Compile();
 
-	static string GetMemberName<T>(in Expression<T> expression) => expression.Body switch
+	static string? GetMemberPathOrNullForCapturedValue<T>(in Expression<T>? expression)
 	{
-		MemberExpression m => m.Member.Name,
-		UnaryExpression { Operand: MemberExpression m } => m.Member.Name,
-		_ => throw new InvalidOperationException("Invalid getter. The `getter` parameter must point directly to a property in the ViewModel and cannot add additional logic")
-	};
+		if (expression is null)
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/TypedGesturesExtensions.cs]: Fixed in `03964ad` by extracting the shared expression parsing/member-path logic into `ExpressionPathHelpers` and updating typed gestures to use it instead of maintaining a duplicate copy.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/TypedBindingExtensions.cs [hunk 10]
```diff
@@ -250,12 +256,57 @@ public static TBindable Bind<TBindable, TBindingContext, TSource, TParam, TDest>
...
+
+		return string.Join(".", handlers.Select(static handler => handler.Item2));
+	}
+
+	static string? GetMemberPathOrNullForCapturedValue<T>(in Expression<T> expression)
+	{
+		ArgumentNullException.ThrowIfNull(expression);
+
+		var members = new Stack<string>();
+		var currentExpression = UnwrapConvertExpression(expression.Body);
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/TypedBindingExtensions.cs]: Fixed in `03964ad` by extracting the shared expression parsing/member-path logic into `ExpressionPathHelpers` and updating typed bindings to use the shared helper.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/ElementExtensions.cs [hunk 11]
```diff
@@ -1,65 +1,121 @@
...
 	}
 
 	/// <summary>
 	/// Remove Dynamic Resource
 	/// </summary>
-	/// <typeparam name="TBindable"></typeparam>
+	/// <typeparam name="TElement"></typeparam>
 	/// <param name="bindable"></param>
 	/// <param name="properties"></param>
 	/// <returns>Layout without Dynamic Resource</returns>
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/ElementExtensions.cs]: Fixed in `5a813f1` by updating the return docs to say `Element` instead of `Layout` for `RemoveDynamicResources`.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/DynamicResourceHandlerExtensions.cs [hunk 12]
```diff
@@ -15,7 +14,7 @@ public static class DynamicResourceHandlerExtensions
 	/// <param name="key"></param>
 	/// <returns>Layout with added Dynamic Resource</returns>
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/DynamicResourceHandlerExtensions.cs]: Fixed in `5a813f1` by updating the return docs to say `Element` instead of `Layout` for `DynamicResource`.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/DynamicResourceHandlerExtensions.cs [hunk 13]
```diff
@@ -30,7 +29,7 @@ public static TDynamicResourceHandler DynamicResource<TDynamicResourceHandler>(t
 	/// <param name="resources"></param>
 	/// <returns>Layout with added Dynamic Resource</returns>
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/DynamicResourceHandlerExtensions.cs]: Fixed in `5a813f1` by updating the return docs to say `Element` instead of `Layout` for `DynamicResources`.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs [hunk 14]
```diff
@@ -186,17 +196,389 @@ public static TBindable Bind<TBindable, TBindingContext, TSource, TParam, TDest>
...
+				TrackTargetValue(targetUpdateTracker, fallbackTargetValue);
+				return fallbackTargetValue;
+			}
+
+			var targetCulture = culture ?? CultureInfo.CurrentUICulture;
+			var sourceValue = getter(sourceObject);
+			var convertedValue = converter?.Convert(sourceValue, targetType ?? typeof(object), parameter, targetCulture) ?? sourceValue;
+			var targetValue = ApplyTargetValueHandlers(convertedValue, stringFormat, targetNullValue, fallbackValue, targetCulture);
+			TrackTargetValue(targetUpdateTracker, targetValue);
+			return targetValue;
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs:662]: Fixed in `249f04b`: pathless typed getter evaluation now catches the common traversal exceptions and returns the fallback/unset target value instead of throwing during binding evaluation. I also added a regression test covering a null path in a command parameter getter without handlers.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs [hunk 15]
```diff
@@ -46,25 +54,27 @@ public static TBindable BindCommand<TBindable, TCommandBindingContext, TParamete
...
 			setter,
 			commandBindingMode,
 			source: source);
 
 		if (parameterGetter is not null)
 		{
-			ArgumentNullException.ThrowIfNull(parameterHandlers);
+			var resolvedParameterBindingMode = parameterHandlers is null && parameterSetter is null && parameterBindingMode is BindingMode.Default
+				? BindingMode.OneTime
+				: parameterBindingMode;
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs:74]: Fixed in `249f04b`: `BindCommand` now fails fast with `ArgumentNullException` for `parameterHandlers` when a parameter setter is supplied without handlers, and a regression test covers that validation.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs [hunk 16]
```diff
@@ -186,17 +201,400 @@ public static TBindable Bind<TBindable, TBindingContext, TSource, TParam, TDest>
...
+		{
+			var convertedValue = converter?.ConvertBack(targetValue, typeof(TSource), converterParameter, System.Globalization.CultureInfo.CurrentUICulture) ?? targetValue;
+			if (ReferenceEquals(convertedValue, BindableProperty.UnsetValue) || ReferenceEquals(convertedValue, Binding.DoNothing))
+			{
+				sourceValue = default;
+				return false;
+			}
+
+			sourceValue = convertedValue is null ? default : ConvertTargetValue<TSource>(convertedValue);
+			return true;
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs]: Fixed in `ab03bc9`: `ConvertBack` returning `null` now skips write-back for non-nullable value-type sources instead of coalescing back to the target value, while nullable/reference source types can still receive `null`. Added a regression test for the non-nullable value-type case.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs [hunk 17]
```diff
@@ -186,17 +201,408 @@ public static TBindable Bind<TBindable, TBindingContext, TSource, TParam, TDest>
...
+		TargetUpdateTracker? targetUpdateTracker,
+		Action applyBinding)
+		where TBindingContext : class?
+	{
+		return (_, args) =>
+		{
+			if (!string.Equals(args.PropertyName, targetProperty.PropertyName, StringComparison.Ordinal))
+			{
+				return;
+			}
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs]: Fixed by treating null/empty `PropertyChangedEventArgs.PropertyName` as an all-properties notification before filtering by the target property. I also added a regression test for targets that only raise the all-properties notification.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/ExpressionPathHelpers.cs [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+		{
+			ParameterExpression when members.Count > 0 => string.Join(".", members),
+			ConstantExpression when members.Count > 0 => null,
+			null when members.Count > 0 => null,
+			_ => throw CreateInvalidGetterException()
+		};
+	}
+
+	public static InvalidOperationException CreateInvalidGetterException()
+		=> new("Invalid getter. The `getter` parameter must point directly to a property in the ViewModel and cannot add additional logic");
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/ExpressionPathHelpers.cs]: Updated the exception message to say the getter must point to a property path, which matches the existing nested-path support.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/ExpressionPathHelpers.cs [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+	{
+		ArgumentNullException.ThrowIfNull(handlers);
+
+		if (handlers.Length is 0 || handlers.Any(static handler => string.IsNullOrWhiteSpace(handler.Item2)))
+		{
+			throw CreateInvalidGetterException();
+		}
+
+		return string.Join(".", handlers.Select(static handler => handler.Item2));
+	}
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/ExpressionPathHelpers.cs:29]: Addressed by using `ArgumentException` with `ParamName == "handlers"` for invalid handler arrays/member names, and updating the public binding tests to assert that behavior.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/ExpressionPathHelpers.cs [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+
+	public static string GetMemberPath<TBindingContext>((Func<TBindingContext, object?>, string)[] handlers)
+	{
+		ArgumentNullException.ThrowIfNull(handlers);
+
+		if (handlers.Length is 0)
+		{
+			throw new ArgumentException("The `handlers` parameter must include at least one member path segment.", nameof(handlers));
+		}
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/ExpressionPathHelpers.cs:21]: Aligned this back to the public typed-binding expectation: empty handler arrays now throw `InvalidOperationException`, while keeping the handler-specific message.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/ExpressionPathHelpers.cs [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+
+		if (handlers.Any(static handler => string.IsNullOrWhiteSpace(handler.Item2)))
+		{
+			throw new ArgumentException("The `handlers` parameter cannot contain null/empty member names.", nameof(handlers));
+		}
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/ExpressionPathHelpers.cs:26]: Aligned whitespace/null handler names to throw `InvalidOperationException` as well, with the existing tests restored to assert that public API behavior.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

**Comment by TheCodeTraveler** [comment]
Hey Gerald! Is this ready for review? Please convert it to a draft PR if you are still working on it.

**Comment by jfversluis** [comment]
> Hey Gerald! Is this ready for review? Please convert it to a draft PR if you are still working on it.

Yes, this is ready for review and no longer needs to be a draft. The latest review feedback has been addressed and the branch is up to date.

File: src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs [hunk 22]
```diff
@@ -155,18 +170,18 @@ public static TBindable Bind<TBindable, TBindingContext, TSource, TParam, TDest>
 			_ => new FuncConverter<TSource, TDest, TParam>(convert, convertBack)
 		};
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/TypedBindingsExtensions.Handlers.cs]: Fixed by replacing the typed-binding inline `FuncConverter` usage with a typed-binding wrapper where a missing `convert` passes the source value through and a missing `convertBack` returns `Binding.DoNothing`. I also extended the existing handler-overload tests for convert-only and convertBack-only behavior.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

File: src/CommunityToolkit.Maui.Markup/TypedBindingExtensions.cs [hunk 23]
```diff
@@ -198,17 +198,22 @@ public static TBindable Bind<TBindable, TBindingContext, TSource, TParam, TDest>
...
 
 	{
-		var getterFunc = ConvertExpressionToFunc(getter);
+		var getterFunc = ExpressionPathHelpers.ConvertExpressionToFunc(getter);
+		var path = ExpressionPathHelpers.GetMemberPathOrNullForCapturedValue(getter);
+		var converter = (convert, convertBack) switch
+		{
+			(null, null) => null,
+			_ => new FuncConverter<TSource, TDest, TParam>(convert, convertBack)
+		};
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/CommunityToolkit.Maui.Markup/TypedBindingExtensions.cs]: Fixed the expression overload to use the same typed-binding wrapper, so convertBack-only bindings populate the target from the source and convert-only bindings skip source write-back instead of writing defaults. The expression-overload tests now cover both cases.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

**Review by TheCodeTraveler — APPROVED** [review]
Thanks Gerald!
